### PR TITLE
Operations for optional values

### PR DESCRIPTION
**Pull request by Horneth**
Closes https://github.com/broadinstitute/cromwell/issues/1765

### CODE REVIEW

**Comment by cjllanwarne** [comment]
Should we double check the semantics of this?

E.g. as it currently is, we fail the workflow if we try to do (e.g.) 
```
Int? x  // unsupplied
Int y = x + 5
```

But I could imagine what we really want to do is map the Options, e.g.:
```
Int? x
Int? y = x + 5
```

**Comment by Horneth** [comment]
Yes we should definitely talk about it, I should have put the request for feedback label.

File: src/main/scala/wdl4s/values/WdlString.scala [hunk 1]
```diff
@@ -12,21 +12,29 @@ case class WdlString(value: String) extends WdlPrimitive {
...
     case r: WdlString => Success(WdlBoolean(value == r.value))
+    case WdlOptionalValue(_, Some(r)) => equals(r)
+    case r: WdlOptionalValue => emptyValue(r)
     case _ => invalid(s"$value == $rhs")
   }
 
   override def lessThan(rhs: WdlValue): Try[WdlBoolean] = rhs match {
     case r: WdlString => Success(WdlBoolean(value < r.value))
+    case WdlOptionalValue(_, Some(r)) => lessThan(r)
+    case r: WdlOptionalValue => emptyValue(r)
```
Comment (cjllanwarne) [inline src/main/scala/wdl4s/values/WdlString.scala]: I'm sure there's some clever way to DRY all these cases, e.g.
```
def operatorIfSet[A, B](wdlOptional: WdlOptionalValue, ifSet: A=> B) = wdlOptional match {
  case WdlOptionalValue(_. Some(a)) => ifSet(a)
  else emptyValue(wdlOptional)
}
```
and then in this situation:
```
override def lessThan[...] {
  [...]
  case o: WdlOptionalValue => operatorIfSet(o, lessThan _)
```

**Comment by cjllanwarne** [comment]
Otherwise, LGTM 👍

**Comment by mcovarr** [comment]
:+1:

**Pull request merged**